### ISSUE TITLE
[BUG] Lazy loaded routes are not documented

### ISSUE THREAD

**Issue by moramike** [compodoc/compodoc#954]
##### **Overview of the issue**

Lazy Loaded routes are not showing in documentation:

![image](https://user-images.githubusercontent.com/32961675/81209339-06ca2980-8f96-11ea-92ab-a7c8c631a13e.png)

##### **Operating System, Node.js, npm, compodoc version(s)**

OS: Windows
Node: 12.8.0
npm: 6.10.2
Angular CLI: 9.1.0

##### **Compodoc installed globally or locally ?**

Tried both

##### **If possible sourcecode of the file where it breaks**

Here is part of my `app-routing.module.ts`
```
// src/app/app-routing.module.ts
const routes: Routes = [
  {
    path: 'module1',
    loadChildren: () =>
      import('./modules/module1/module1.module').then((m) => m.Module1Module),
  }
]
```

And for each module, such as `Module1`:
```ts
// src/app/modules/module1/module1-routing.module.ts
const routes: Routes = [
  {
    path: '',
    component: Module1Component,
    data: {
      ...
    },
    canActivate: [AuthGuard],
    children: [
      {
        path: '',
        redirectTo: 'list'
      },
      {
        path: 'list',
        pathMatch: 'full',
        component: ListComponent,
        data: {
          ...
        },
      },
      {
        path: ':id',
        component: DetailsComponent,
        data: {
          ...
        },
      },
    ],
  },
];
```

##### **If possible your terminal logs before the error**

```console
Error during generation of routes JSON file, maybe a trailing comma or an external variable inside one route.
```

![image](https://user-images.githubusercontent.com/32961675/81209729-848e3500-8f96-11ea-97a2-8c8d995c1a64.png)

##### **Motivation for or Use Case**

It should document all routes in the corresponding section.

##### **Reproduce the error**

- Create a new app using the CLI and adding `--routing`
- Create a new module (`ng g m module --routing`)
- Add some routes an sample components as previously specified.
- Try to create the documentation (`npx compodoc -p tsconfig.json`)

##### **Related issues**

Several times, but all of them have been closed due to inactivity
#690, #569, #609, #805, #819.

##### **Suggest a Fix**

I haven't found a solution yet.

**Comment by fbaba-nibtravel** [comment]
I'm having the same issue. It feels like the routes don't support the new syntax for lazy loading. 

If that's the case, is there anyone looking into this?

**Comment by diegovincent** [comment]
I offer myself to work on this, but I am available starting next week.

**Comment by ikumaran** [comment]
I too need this bug fixed. Thanks @diegovincent to get started.

**Comment by cwistedt** [comment]
same problem here.

**Comment by felixlindemann** [comment]
Same Problem here - I upgraded to Angular 9 - now no routes will be included into the documentation at all

I get an error with asm.js (don't know, if it is related)

`(node:14104) V8: C:\repos\myProject\webapp\node_modules\viz.js\viz.js:33 Invalid asm.js: Function definition doesn't match use
(Use `node --trace-warnings ...` to show where the warning was created)`

**Comment by jgomesmv** [comment]
I have the same problem no routes presented.
![image](https://user-images.githubusercontent.com/8100887/91730387-be1a2a00-eb9d-11ea-841a-17b8ae5a6f21.png)

But even when I remove all lazy loading logic the same error is shown: 
![image](https://user-images.githubusercontent.com/8100887/91732009-890ed700-eb9f-11ea-8723-43f0fc111736.png)

I'm using this dependencies:
![image](https://user-images.githubusercontent.com/8100887/91732112-b196d100-eb9f-11ea-9748-64b0b44eb2e5.png)

**Comment by codingsamuel** [comment]
getting this error as well on angular 10.1.1 and compodoc 1.1.11

is there any progress on this?

**Comment by rodmax** [comment]
Hi there. I getting the same issue and what i see.

TL;DR; globally installed compodoc, seems works well (for me)

### :red_circle: compodoc(version 0.0.41) installed as `devDependency`
![image](https://user-images.githubusercontent.com/6079597/95721182-e310bb00-0c7a-11eb-8ddc-93af67557bad.png)

```bash
npm run compodoc
```
![image](https://user-images.githubusercontent.com/6079597/95721349-210ddf00-0c7b-11eb-9a0f-0de96a548b61.png)


### :green_circle: compodoc(v0.0.41) installed as GLOBAL package ("npm install -g compodoc")
All works without router parsing errors !!! and documentation looks good
```bash
compodoc -p tsconfig.json src 
```

![image](https://user-images.githubusercontent.com/6079597/95721683-9da0bd80-0c7b-11eb-9290-a9b7bd03f672.png)


So it seems some shared local dependency or special working mode of compodoc as devDepency add side effect

**Comment by headonkeyboard** [comment]
Hi,
I've just found a workaround that works for me on compodoc 1.1.11. Just replace

`(m) => m.MyModule`

by 

`m => m.MyModule`

**Comment by christian-fries** [comment]
I was always using the global package, still it did not work. 

But I can confirm the workaround posted by @headonkeyboard

**Comment by NurGuz** [comment]
> I was always using the global package, still it did not work.
> 
> But I can confirm the workaround posted by @headonkeyboard

Same here

**Comment by alastair-todd** [comment]
> Routes parsing error, maybe a trailing comma or an external variable ?

Same for me as @rodmax

**Comment by saidsoualhi** [comment]
I've tried the workaround by @headonkeyboard , but it's not working for me
Angular version: 11.2.0
Compodoc version: 1.1.11

**Comment by eugeniucarvalho** [comment]
I've used the workaround provided by @headonkeyboard and it work for me.
My versions are:
Angular : 11.2.9
Compodoc: 1.1.11

**Comment by ChangAlanCL** [comment]
Beg a pardon, I'm not sure which file @headonkeyboard is referring to?

**Comment by joxy379** [comment]
The issue still exists.

**Comment by asennyey** [comment]
For anyone finding this later than the above thread, it is still kind of an error if you use prettier with trailingComma set to "es5", the parser is assuming that the javascript object will be proper JSON.
To fix this, set trailingComma to none.

**Comment by CarloConflitti** [comment]
The issue still exists.

Angular 14.2.12
Compodoc 1.1.19

**Comment by Mahmoudibrahim03** [comment]
The issue still exists.

Angular 15.2.0
Compodoc 1.1.19

**Comment by plusnia-pgs** [comment]
Does anyone solve this problem?

**Comment by dontboyle** [comment]
Still an issue *bump* thread
Angular 15.2.0
Compodoc 1.1.19

**Comment by straiforos** [comment]
- This is still an issue bumping thread, Angular 13, compodoc 1.1.19

**Comment by russelporosky** [comment]
Confirming still an issue, Angular 19, compodoc 1.1.26, and using standalone components.

**Comment by straiforos** [comment]
I attempted to fix it through debugging but could not find the root cause. 

https://github.com/compodoc/compodoc/issues/1405

Closed my issue since it never got traction.
